### PR TITLE
Add instructions on how to setup mailing on Autolab

### PR DESCRIPTION
**Pull request by victorhuangwq**
## Description

This PR centralizes the settings that needs to be changed in the `production.rb` file (as it was scattered over a two files previously). At the same time, it adds documentation to our docs on what needs to be changed and how. It also deprecates sendmail as the default mode of sending mail using Autolab.

## Motivation and Context

Users desire a simple and clear way to setup mailing with Autolab such that they can register users.

## How Has This Been Tested?

This settings has been tested using a mailgun account locally, by following the exact steps on the documentation instruction. 
After the mailing server has been set-up, I registered an account using my personal email, and I received the registration email.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [x] I have updated the documentation accordingly, included in this PR

## Other issues / help required

As of now, this method of updating the production.rb works with the Docker installation method as well. However, from discussion we do see a more elegant alternative using docker-compose, and there will be a follow-up PR.

### CODE REVIEW

File: docs/installation/mailing.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Mailing Setup
+
+Autolab requires mailing to allow users to register accounts and reset passwords. You will also be able to make announcements through Autolab as well. The recommended approach is to setup Autolab to use a SMTP server, such as [mailgun](https://mailgun.com), [SendGrid](https://sendgrid.com), [Amazon SES](https://aws.amazon.com/ses/) or any other valid SMTP mail servers to send out email.
+
+We intend this instructions mainly for production usage. To set Autolab up to use a custom SMTP Server, edit the following in `production.rb` that you have created. (If you would like to test it in development, add the following settings into `development.rb`).
```
Comment (ashleyzhang) [inline docs/installation/mailing.md:5]: nit: Maybe specify that `production.rb` and `development.rb` are in config/environments?

**Review by ashleyzhang — APPROVED** [review]
Left a minor suggestion but works great! Tested with a sendgrid account locally and got the following email as expected when registering an account. 
Screen Shot 2021-04-30 at 2 49 54 PM

File: docs/installation/mailing.md (hunk 1 above)
Comment (victorhuangwq) [reply docs/installation/mailing.md]: thanks for the suggestion :). change has been made

**Review by ashleyzhang — APPROVED** [review]
LGTM!

**Pull request merged**